src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1788,14 +1789,28 @@ public DatabaseRecord ReadDatabaseRecord()
 
         private void OnCertificateChange(object sender, EventArgs e)
         {
-            if (_requestExecutor.IsValueCreated == false)
+            if (_proxyRequestExecutor.IsValueCreated == false)
                 return;
 
-            using (_requestExecutor.Value)
-                _requestExecutor = CreateRequestExecutor();
+            using (_proxyRequestExecutor.Value)
+                _proxyRequestExecutor = CreateRequestExecutor();
         }
 
-        private Lazy<RequestExecutor> CreateRequestExecutor() => new(() => RequestExecutor.CreateForServer(new[] { ServerStore.Configuration.Core.GetNodeHttpServerUrl(ServerStore.Server.WebUrl) }, Name, ServerStore.Server.Certificate.Certificate, DocumentConventions.DefaultForServer), LazyThreadSafetyMode.ExecutionAndPublication);
+        public readonly DocumentConventions DocumentConventionsForProxy = new()
+        {
+            SendApplicationIdentifier = DocumentConventions.DefaultForServer.SendApplicationIdentifier,
+            MaxContextSizeToKeep = DocumentConventions.DefaultForServer.MaxContextSizeToKeep,
+            HttpPooledConnectionLifetime = DocumentConventions.DefaultForServer.HttpPooledConnectionLifetime,
+            HttpClientType = typeof(DocumentDatabase),
+            DisableTopologyCache = DocumentConventions.DefaultForServer.DisableTopologyCache,
+            DisposeCertificate = DocumentConventions.DefaultForServer.DisposeCertificate,
+            IncludePromotablesInTopology = true
+        };
+
+        private Lazy<RequestExecutor> CreateRequestExecutor() =>
+            new(
+                () => RequestExecutor.CreateForServer(new[] {ServerStore.Configuration.Core.GetNodeHttpServerUrl(ServerStore.Server.WebUrl)}, Name,
```

Comment:
what about shards?

Reply:
I think we probably don't want promotables for shard executors because we only want to reach promotables via selected node (which is only proxy)

Reply:
Added support for sharded dbs